daft/expressions/expressions.py

```diff
@@ -2922,6 +2922,17 @@ def join(self, delimiter: str | Expression) -> Expression:
         delimiter_expr = Expression._to_expression(delimiter)
         return Expression._from_pyexpr(native.list_join(self._expr, delimiter_expr._expr))
 
+    # todo: do we want type to be a Map expression? how should we do this?
```

Comment:
correct me if I am right but can't we return something like 

https://github.com/Eventual-Inc/Daft/blob/6e0944a5017f1c5b91451eb07c006ab85581d276/daft/expressions/expressions.py#L3072

Reply:
Ah great question -- not really because then we'd lose access to the other Expression functionality such as `Expression.is_null()`.

There generally isn't a great solution here given the dynamic nature of how we're constructing these Expressions tbh. We can only definitively know what output type an Expression would produce once it is resolved against a schema. For example:

```
col("a") + col("b")
```

Should this expression produce a int32 type? The answer is: it depends. Because if col("a") and col("b") were string types, and we resolve this expression against that schema, then the output type will be a string.